cime/tools/mapping/gen_domain_files/src/gen_domain.F90

```diff
@@ -335,9 +346,14 @@ subroutine gen_domain(fmap, fn1_out_ocn, fn2_out_lnd, fn2_out_ocn, set_fv_pole_y
 
       if (.not. complf) then
 
-         ! Determine ocn mask on ocn grid
-         call check_ret(nf_inq_varid(fid,'mask'//trim(suffix), vid ))
-         call check_ret(nf_get_var_int   (fid,vid,omask ))
+         ! Determine ocn mask on ocn grid; if mask is not present, we assume
+         ! that the grid covers the entire domain, so mask = 1 for all points.
+         if (var_exists(fid, 'mask'//trim(suffix))) then
+            call check_ret(nf_inq_varid(fid,'mask'//trim(suffix), vid ))
+            call check_ret(nf_get_var_int   (fid,vid,omask ))
+         else
+            omask(:) = 1
+         end if
```

Comment:
Bill, after talking about this with CESM, can you implement this differently?   Add another optional argument called something like "--noomask".  If its present AND the mask variable does not exist, then set omask to 1.   Error if --nomask is set AND the variable DOES exist.  Or if the variable is doesn't exist and --nomask is not set.  Similar for mask_a below.

Reply:
maybe "--setomask" and "--setamask" would be better names.

Reply:
Rob, what about “—no-omask” and “—no-amask”, and if unset and mask does not exist, throw an error, but IF SET, then read masks if they exist and set equal to 1 if they do not? Is there a reason to throw an error if the flag is set and the mask actually already exists? I get wanting to throw an error by default when the flag is unset and the masks don’t exist, that totally makes sense.

Reply:
The if test is "do something if mask is not present".  So we instead want an option which specifies that action.  The action to take is "set the mask".  So the option should be "set-omask"  "set-amask".

Yeah I guess you don't have to flag an error if the flag is set and the mask is present.

Reply:
I have this implemented, just need to test. But...should we just omit this addition and either close this PR, or just include the fminval and fmaxval additions and raise an error if mask doesn’t exist? Now that Paul made the changes in tempest to propagate mask and domain, we shouldn’t need to worry about mask not being present.